services/QuillLMS/client/app/bundles/Staff/components/comprehension/activitySessions/promptTable.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import stripHtml from "string-strip-html";
+
+import { DataTable, Spinner } from '../../../../Shared/index';
+import { PROMPT_ATTEMPTS_FEEDBACK_LABELS, PROMPT_HEADER_LABELS } from '../../../../../constants/comprehension';
+import { ActivityInterface, PromptInterface } from "../../../interfaces/comprehensionInterfaces";
+
+interface PromptTableProps {
+  activity: ActivityInterface;
+  prompt: PromptInterface;
+  showHeader?: boolean;
+}
+const PromptTable = ({ activity, prompt, showHeader }: PromptTableProps) => {
+
+
+  function formatFirstTableData(prompt: any) {
+    const { attempts } = prompt;
+    const keys = attempts && Object.keys(attempts);
+    let completed: boolean;
+    if(keys.length === 5) {
+      completed = true;
+    } else {
+      const usedAttempts = [];
+      keys.forEach(key => {
+        attempts[key].forEach((attempt: any) => {
+          if(attempt.used) {
+            usedAttempts.push(attempt);
+          }
+        });
+      });
+      completed = !!usedAttempts.some((attempt) => attempt.optimal);
+    }
+    return {
+      attemptsLabel: 'Attempts',
+      attemptsValue: keys.length,
+      completedLabel: 'Completed',
+      completedValue: completed ? 'True' : 'False'
+    }
+  }
+
+  function formatFeedbackData(prompt: any) {
+    const { attempts, prompt_id } = prompt;
+    const { prompts } = activity;
+    const matchedPrompt = prompts.filter(prompt => prompt.id === prompt_id)[0];
+    const keys = attempts && Object.keys(attempts);
+    const rows = [];
+    keys.map(key => {
+      const filteredAttempt = attempts[key].filter(attempt => attempt.used)[0];
+      const attempt = filteredAttempt || attempts[key][0];
+      const { entry, feedback_text, feedback_type, optimal } = attempt;
+      const { attemptLabel, feedbackLabel } = PROMPT_ATTEMPTS_FEEDBACK_LABELS[key];
+      const attemptObject: any = {};
+      const feedbackObject: any = {};
+      attemptObject.status = attemptLabel;
+      attemptObject.results = (<div>
+        <b>{matchedPrompt && matchedPrompt.text}</b>
+        <p className="entry">{entry}</p>
+      </div>);
+      feedbackObject.status = feedbackLabel;
+      feedbackObject.results = stripHtml(feedback_text);
+      feedbackObject.feedback = feedback_type;
+      feedbackObject.className = optimal ? 'optimal' : 'sub-optimal'
```

Comment:
I think it might be clearer to define the objects like this:

```
const attemptObject = {
    status: attemptLabel,
    results: ...
}

const feedbackObject = {
   ...
}
```

Rather than defining them as empty and then immediately adding attributes.

Reply:
updating!